src/pages/MovieDetails.js

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+
+const MovieDetails = () => {
+  const router = useRouter();
+  const { id } = router.query;
+  const [movie, setMovie] = useState(null);
+
+  useEffect(() => {
+    if (id) {
+      fetch(`http://localhost:3001/movies/${id}`)
+        .then((response) => response.json())
+        .then((data) => setMovie(data))
+        .catch((error) => console.error('Error fetching movie details:', error));
+    }
+  }, [id]);
+
+  if (!movie) {
+    return <div>Loading...</div>;
+  }
```

Comment:
At least use https://mui.com/material-ui/react-progress/ instead 🤔

Reply:
Fixed for moviedetails, and actordetails 🥳